types/src/namespaced_data.rs

```diff
@@ -32,18 +32,109 @@ pub const NAMESPACED_DATA_ID_CODEC: u64 = 0x7820;
 /// [`ExtendedDataSquare`]: crate::rsmt2d::ExtendedDataSquare
 #[derive(Debug, PartialEq, Clone, Copy)]
 pub struct NamespacedDataId {
+    /// Index of the row in the [`ExtendedDataSquare`].
+    ///
+    /// [`ExtendedDataSquare`]: crate::rsmt2d::ExtendedDataSquare
+    pub row: RowId,
     /// A namespace of the [`Share`]s.
     ///
     /// [`Share`]: crate::Share
     pub namespace: Namespace,
-    /// Index of the row in the [`ExtendedDataSquare`].
+}
+
+/// `NamespacedData` is constructed out of the ExtendedDataSquare, containing up to a row of
+/// shares belonging to a particular namespace and a proof of their inclusion. If, for
+/// particular EDS, shares from the namespace span multiple rows, one needs multiple
+/// NamespacedData instances to cover the whole range.
```

Comment:
For docs, please go with first short title what is what and then add an explanation in a body below. [rustdoc guide](https://doc.rust-lang.org/rustdoc/how-to-write-documentation.html#documenting-components)

```suggestion
/// `NamespacedData` contains up to a row of shares belonging to a particular namespace and a proof of their inclusion.
///
/// It is constructed out of the ExtendedDataSquare. If, for particular EDS, shares from the namespace span multiple rows,
/// one needs multiple NamespacedData instances to cover the whole range.
```

Reply:
I don't know if thats well formatted :facepalm: